src/components/containers/portal.js

```diff
@@ -0,0 +1,53 @@
+import { Component } from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+
+// Use a ternary operator to make sure that the document object is defined
+const portalRoot =
+    typeof document !== `undefined` ? document.getElementById('modal') : null
```

Comment:
can we add a `const has_document = typeof document !== `undefined` and reuse it?